sdk/communication/azure-communication-identity/azure/communication/identity/_shared/models.py

```diff
@@ -320,7 +324,7 @@ def identifier_from_raw_id(raw_id: str) -> CommunicationIdentifier:
     if raw_id.startswith(PHONE_NUMBER_PREFIX):
         return PhoneNumberIdentifier(
             value=raw_id[len(PHONE_NUMBER_PREFIX) :], raw_id=raw_id
-        )
+        )  # type: ignore
```

Comment:
What is the reason for the ignores in this function? It feels like there might be something we need to fix

Reply:
@kristapratico Could you please help me with the solution for this mypy error ?

Reply:
Took a closer look at this and I don't think the protocol defined works for these types. The more difficult part is that each identifier has a unique TypedDict which is incompatible with a Mapping (you can read more [here](https://github.com/python/mypy/issues/4976)). I think fixing this will take a some brainstorming + maybe a refactor and is probably better for a different PR. Can you open an issue to take another look at the protocol for this?

Reply:
Created an issue for this : https://github.com/Azure/azure-sdk-for-python/issues/31098